osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    public class OsuModAlternate : ModAlternate<OsuHitObject, OsuAction>
+    {
+        private OsuAction? lastActionPressed;
+
+        protected override void ResetActionStates()
+        {
+            lastActionPressed = null;
+        }
+
+        protected override bool OnPressed(OsuAction action)
+        {
+            if (IsBreakTime.Value)
+                return false;
+
+            if (HighestCombo.Value == 0)
```

Comment:
why?

Reply:
Highest combo is based on the number of non-miss hits. It is possible to do `ScoreProcessor.GetStatistic()` multiple times for each hit result type, this is much simpler.

Reply:
I can't parse this comment. What does this aim to do exactly? Not process inputs until the first user hit? I'd actually think using just `Combo` would be more correct, to allow recovery from any position after a miss.